zigup.zig

```diff
@@ -205,6 +205,23 @@ pub fn main2() !u8 {
                 help();
                 return 0;
             } else {
+                if (newlen == 0 and std.mem.eql(u8, "run", arg)) {
+                    const version_string = args[i + 1];
+                    const install_dir_string = try getInstallDir(allocator, .{ .create = true });
+                    defer allocator.free(install_dir_string);
+
+                    const compiler_dir = try std.fs.path.join(allocator, &[_][]const u8{ install_dir_string, version_string });
+                    defer allocator.free(compiler_dir);
+
+                    var argv = std.ArrayList([]const u8).init(allocator);
+                    try argv.append(try std.fs.path.join(allocator, &.{ compiler_dir, "files", "zig" ++ builtin.target.exeFileExt() }));
+                    try argv.appendSlice(args[i + 2 ..]);
+
+                    const proc = try std.ChildProcess.init(argv.items, allocator);
+                    defer proc.deinit();
+                    const ret_val = try proc.spawnAndWait();
+                    return ret_val.Exited;
+                }
```

Comment:
Let's put this in a function elsewhere to keep the command-line parsing section shorter.  I think this would make it easier to follow what's going on.

Reply:
Done :+1: